phpBB/phpbb/install/console/command/update/update.php

```diff
@@ -0,0 +1,176 @@
+	 */
+	protected $install_helper;
+
+	/**
+	 * @var language
+	 */
+	protected $language;
+
+	/**
+	 * Constructor
+	 *
+	 * @param language			$language
+	 * @param factory			$factory
+	 * @param installer			$installer
+	 * @param install_helper	$install_helper
+	 */
+	public function __construct(language $language, factory $factory, installer $installer, install_helper $install_helper)
+	{
+		$this->iohandler_factory = $factory;
+		$this->installer = $installer;
+		$this->language = $language;
+		$this->install_helper = $install_helper;
+
+		parent::__construct(new \phpbb\user($language, 'datetime'));
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	protected function configure()
+	{
+		$this
+			->setName('update')
+			->addArgument(
+				'config-file',
+				InputArgument::REQUIRED,
+				$this->language->lang('CLI_CONFIG_FILE'))
+			->setDescription($this->language->lang('CLI_UPDATE_BOARD'))
+		;
+	}
+
+	/**
+	 * Executes the command update.
+	 *
+	 * Install the board
+	 *
+	 * @param InputInterface  $input  An InputInterface instance
+	 * @param OutputInterface $output An OutputInterface instance
+	 *
+	 * @return int
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output)
+	{
+		$this->iohandler_factory->set_environment('cli');
+
+		/** @var \phpbb\install\helper\iohandler\cli_iohandler $iohandler */
+		$iohandler = $this->iohandler_factory->get();
+		$style = new SymfonyStyle($input, $output);
+		$iohandler->set_style($style, $output);
+
+		$this->installer->set_iohandler($iohandler);
+
+		$config_file = $input->getArgument('config-file');
+
+		if (!$this->install_helper->is_phpbb_installed())
+		{
+			$iohandler->add_error_message('INSTALL_PHPBB_NOT_INSTALLED');
+
+			return 1;
+		}
+
+		if (!is_file($config_file))
+		{
+			$iohandler->add_error_message(array('MISSING_FILE', $config_file));
+
+			return 1;
+		}
+
+		try
+		{
+			$config = Yaml::parse(file_get_contents($config_file), true, false);
+		}
+		catch (ParseException $e)
+		{
+			$iohandler->add_error_message(array('INVALID_YAML_FILE', $config_file));
+
+			return 1;
+		}
+
+		$processor = new Processor();
+		$configuration = new updater_configuration();
+
+		try
+		{
+			$config = $processor->processConfiguration($configuration, $config);
+		}
+		catch (Exception $e)
+		{
+			$iohandler->add_error_message('INVALID_CONFIGURATION', $e->getMessage());
+
+			return 1;
+		}
+
+		$this->register_configuration($iohandler, $config);
+
+		try
+		{
+			$this->installer->run();
+		}
+		catch (installer_exception $e)
+		{
+			$iohandler->add_error_message($e->getMessage());
+			return 1;
+		}
+	}
+
+	/**
+	 * Register the configuration to simulate the forms.
+	 *
+	 * @param cli_iohandler $iohandler
+	 * @param array $config
+	 */
+	private function register_configuration(cli_iohandler $iohandler, $config)
+	{
+		$iohandler->set_input('update_type', $config['type']);
+		$iohandler->set_input('submit_update', 'submit');
+
+		$iohandler->set_input('method', '.tar');
```

Comment:
it's just to have something valid (tar is always available). But it will not be used: we are in CLI, you can't have a link to DL the archive. Reminds me I still need to check what will happen in this case